src/spec/expectations.cr

```diff
@@ -170,7 +170,7 @@ module Spec
     end
 
     def match(actual_value)
-      actual_value =~ @expected_value
+      actual_value.responds_to?(:=~) && actual_value =~ @expected_value
```

Comment:
Not really a fix: you moved the "returns nil" logic from global operators to a specific use case.

The fact that the match expectation *is* called with non String nor RegExp objects may be proof that the global behavior is actually good?